Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,487 @@
+private noncomputable def F' (n : ℕ) (f : ℂ → ℂ) := fun z ↦ F f z • exp ((z^2-1) * (n : ℝ)⁻¹)
+
+
+-- Small lemma : Sup of abs is nonneg
+lemma sSupAbsIm_nonneg (f : ℂ → ℂ) (x : ℝ) : 0 ≤ sSupAbsIm f x := by
+  apply Real.sSup_nonneg
+  rintro y ⟨z1, _, hz2⟩
+  simp only [← hz2, comp, map_nonneg]
+
+
+-- Definition rewrites for invInterpStrip
+lemma invInterpStrip_eq_of_pos (f : ℂ → ℂ ) (h0 : 0 < sSupAbsIm f 0) (h1 : 0 < sSupAbsIm f 1) :
+    invInterpStrip f z = (sSupAbsIm f 0)^(z-1) * (sSupAbsIm f 1)^(-z) := by
+  simp only [ne_of_gt h0, ne_of_gt h1, invInterpStrip, if_false, or_false]
+
+
+lemma invInterpStrip_eq_of_zero (f : ℂ → ℂ ) (h : (sSupAbsIm f 0) = 0 ∨ (sSupAbsIm f 1) = 0) :
+    invInterpStrip f z = 0 :=
+  if_pos h
+
+-- Differentiable continuous function invInterpStrip
+lemma diffContOnCl_invInterpStrip (f : ℂ → ℂ) : DiffContOnCl ℂ (invInterpStrip f) (strip 0 1) := by
+  by_cases (sSupAbsIm f 0) = 0 ∨ (sSupAbsIm f 1) = 0
+  -- Case everywhere 0
+  -- Starting with Lemma to handle rewriting of invInterpStrip as a function.
+  · have hzero : invInterpStrip f = 0 := by
+      rw [funext_iff]
+      intro z
+      simp only [invInterpStrip_eq_of_zero z f h, Pi.zero_apply, eq_self_iff_true]
+    rw [hzero, Pi.zero_def]
+    exact diffContOnCl_const
+
+  -- Case nowhere 0
+  · push_neg at h
+    cases' h with h0 h1
+    rw [ne_comm] at h0 h1
+    apply Differentiable.diffContOnCl
+    intro z
+    -- Same strategy.
+    have hpos: invInterpStrip f = fun (z : ℂ) ↦ (sSupAbsIm f 0)^(z-1) * (sSupAbsIm f 1)^(-z) := by
+      funext z
+      rw [invInterpStrip_eq_of_pos z f (lt_of_le_of_ne (sSupAbsIm_nonneg f 0) h0)
+      (lt_of_le_of_ne (sSupAbsIm_nonneg f 1) h1)]
+    rw [hpos]
+    refine DifferentiableAt.mul ?_ ?_
+    · apply DifferentiableAt.const_cpow (DifferentiableAt.sub_const (differentiableAt_id') 1) _
+      left; simp only [Ne.def, ofReal_eq_zero]; rwa [eq_comm]
+    · refine DifferentiableAt.const_cpow ?_ ?_
+      · rw [differentiableAt_neg_iff]
+        apply differentiableAt_id'
+      · left; simp only [Ne.def, ofReal_eq_zero]; rwa [eq_comm]
+
+
+
+lemma diffContOnCl_F' (f : ℂ → ℂ) (n : ℕ) (hd : DiffContOnCl ℂ f (strip 0 1)) :
+    DiffContOnCl ℂ (F' n f) (strip 0 1) := by
+  refine DiffContOnCl.smul (DiffContOnCl.smul hd (diffContOnCl_invInterpStrip f) )
+    <| Differentiable.diffContOnCl <| Differentiable.cexp <| Differentiable.mul ?_
+      (differentiable_const _)
+  rw [differentiable_sub_const_iff]
+  simp only [cpow_two, differentiable_id', Differentiable.pow]
+
+
+
+-- The function `f` is bounded by `sSupAbsIm`
+lemma abs_le_sSupAbsIm (f : ℂ → ℂ) (z : ℂ) (hD : z ∈ (closedStrip 0 1))
+    (hB : BddAbove ((abs ∘ f) '' (closedStrip 0 1))) : abs (f z) ≤ sSupAbsIm f (z.re) := by
+  refine le_csSup ?_ ?_
+  · apply BddAbove.mono (image_subset (abs ∘ f) _) hB
+    apply preimage_mono
+    simp only [singleton_subset_iff]
+    exact hD
+  · apply mem_image_of_mem (abs ∘ f)
+    simp only [mem_preimage, mem_singleton]
+
```

Comment:
Could you maybe use sections with docstrings to organize a little bit your file? Above you have basic properties of `sSupAbsIm`, now you are dealing with properties of strips, then you are going to put all this together. It would help your reader a lot if this structure was clear from the structure of the file.